graceful restart of non-leader server agent makes active job evaluation segfault

### Nomad version
1.5.6

### Operating system and Environment details
amazonlinux:2

### Issue
A gracefull restart of nomad server during deployment fails the deployment/evaluation. This seems like something nomad is supposed to be resilient against?

In my case I noticed it happening when a non-leader failed.  I tested on the leader as well and the same thing happens

### Reproduction steps
restart/reschedule a job
```
$ nomad job restart -reschedule phonetrac_resque
```

While thats doing it's thing, find a non-leader server and:
```
$sudo systemctl restart nomad
```

#### Expected Result
`nomad job run` would keep working?

#### Actual Result
`nomad job run` does not keep working


### Nomad Server logs (if appropriate)
This error popped up in the leader:
```
May 25 21:13:35 ip-10-20-21-12 nomad: 2023-05-25T21:13:35.561Z [ERROR] nomad.raft: failed to appendEntries to: peer="{Nonvoter 66642905-0f42-910f-a356-13350aaef6d0 10.20.11.253:4647}" error=EOF
```

### Nomad Client logs (if appropriate)

```
==> 2023-05-25T21:13:12Z: Restarting 6 allocations
    2023-05-25T21:13:12Z: Rescheduling allocation "cb67e36b" for group "scheduler"
    2023-05-25T21:13:12Z: Allocation "cb67e36b" replaced by "831b16e2", waiting for "831b16e2" to start running
    2023-05-25T21:13:28Z: Allocation "831b16e2" is "running"
    2023-05-25T21:13:28Z: Rescheduling allocation "28b10d02" for group "scheduler"
    2023-05-25T21:13:28Z: Allocation "28b10d02" replaced by "de204fe4", waiting for "de204fe4" to start running
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x240872a]

goroutine 50 [running]:
github.com/hashicorp/nomad/command.(*JobRestartCommand).monitorReplacementAlloc(0xc000aa4280, {0x34e31d8, 0xc0003bb040}, {0xc000251080?, 0xc0009a3860?}, 0x0?)
        github.com/hashicorp/nomad/command/job_restart.go:1045 +0x60a
created by github.com/hashicorp/nomad/command.(*JobRestartCommand).stopAlloc
        github.com/hashicorp/nomad/command/job_restart.go:951 +0x51c
```

--- a/.changelog/17346.txt
+++ b/.changelog/17346.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+cli: Fix a panic in the `nomad job restart` command when monitoring replacement allocations
+```

--- a/command/job_restart.go
+++ b/command/job_restart.go
@@ -1045,7 +1045,7 @@ func (c *JobRestartCommand) monitorReplacementAlloc(
 
 		alloc, qm, err := c.client.Allocations().Info(currentAllocID, q)
 		if err != nil {
-			errCh <- fmt.Errorf("Failed to retrieve allocation %q: %w", limit(alloc.ID, c.length), err)
+			errCh <- fmt.Errorf("Failed to retrieve allocation %q: %w", limit(currentAllocID, c.length), err)
 			return
 		}
 

